x-pack/plugins/actions/server/lib/task_runner_factory.ts

```diff
@@ -128,6 +132,10 @@ export class TaskRunnerFactory {
             `Failed to cleanup action_task_params object [id="${actionTaskParamsId}"]: ${e.message}`
           );
         }
+        // if subApiKey exists, it should be invalidated after usage
```

Comment:
Looking at the flow in this function, seems like there are other places that should invalidate the new API subkey - basically in the sites above where we `throw` - but it may be that some of those do retries, so not sure if they get a new subkey or try to reuse the existing one.  I'm also wondering about clean up of `action_task_params` in some of those cases ...